pennylane_cirq/simulator_device.py

```diff
@@ -136,30 +135,36 @@ def apply(self, operations, **kwargs):
         self.circuit.append(cirq.IdentityGate(len(self.qubits))(*self.qubits))
 
         if self.analytic:
-            self._result = self._simulator.simulate(
-                self.circuit, initial_state=self._initial_state
-            )
+            self._result = self._simulator.simulate(self.circuit, initial_state=self._initial_state)
 
-            self._state = np.array(self._result.state_vector())
+            self._state = self._get_state_from_cirq(self._result)
 
     def analytic_probability(self, wires=None):
         # pylint: disable=missing-function-docstring
         if self._state is None:
             return None
 
         wires = wires or range(self.num_wires)
-        probs = np.abs(self._state) ** 2
+        probs = self._get_computational_basis_probs()
 
         return self.marginal_prob(probs, wires)
 
+    def _get_state_from_cirq(self, result):
+        """Extract the state array from a Cirq TrialResult ``result``"""
+        return np.array(result.state_vector())
```

Comment:
This method has no `self` usage, it can probably be a static method or a class method